core/executor/src/wasm_executor.rs

```diff
@@ -389,24 +423,25 @@ impl_function_executor!(this: FunctionExecutor<'e, E>,
 		if let Some(value) = maybe_value {
 			let value = &value[value_offset as usize..];
 			let written = ::std::cmp::min(value_len as usize, value.len());
-			this.memory.set(value_data, &value[..written]).map_err(|_| UserError("Invalid attempt to set value in ext_get_child_storage_into"))?;
+			this.memory.set(value_data, &value[..written])
+				.map_err(|_| UserError("Invalid attempt to set value in ext_get_child_storage_into"))?;
 			Ok(written as u32)
 		} else {
 			Ok(u32::max_value())
 		}
 	},
 	ext_storage_root(result: *mut u8) => {
 		let r = this.ext.storage_root();
-		this.memory.set(result, r.as_ref()).map_err(|_| UserError("Invalid attempt to set memory in ext_storage_root"))?;
+		this.memory.set(result, r.as_ref())
+			.map_err(|_| UserError("Invalid attempt to set memory in ext_storage_root"))?;
 		Ok(())
 	},
 	ext_child_storage_root(storage_key_data: *const u8, storage_key_len: u32, written_out: *mut u32) -> *mut u8 => {
-		let storage_key = this.memory.get(storage_key_data, storage_key_len as usize).map_err(|_| UserError("Invalid attempt to determine storage_key in ext_child_storage_root"))?;
-		let storage_key = ChildStorageKey::from_slice(&*storage_key)
-			.ok_or_else(||
-				UserError("ext_child_storage_root: child storage key is not valid")
-			)?;
-		let value = this.ext.child_storage_root(storage_key);
+		let storage_key = this.memory.get(storage_key_data, storage_key_len as usize)
+			.map_err(|_| UserError("Invalid attempt to determine storage_key in ext_child_storage_root"))?;
+		let value = this.with_subtrie(&storage_key[..], |this, subtrie|
+			this.ext.child_storage_root(&subtrie)
+		).expect("Called from a valid SubTrie instance");
```

Comment:
I am having some problems with this proof. There is no comment on `with_subtrie` that explains when it can return `Err`.

Reply:
This is related to the comment above `with_subtrie`: with_subtrie should be call with an already resolved subtrie so the query should not fail. This is clearly an unnecessary expect, I will remove it.